src/interfaces/node.cpp

```diff
@@ -105,11 +105,13 @@ class NodeImpl : public Node
             }
 
             // Try to retrieve the CNodeStateStats for each node.
-            TRY_LOCK(::cs_main, lockMain);
-            if (lockMain) {
-                for (auto& node_stats : stats) {
-                    std::get<1>(node_stats) =
-                        GetNodeStateStats(std::get<0>(node_stats).nodeid, std::get<2>(node_stats));
+            if (m_context->peerman) {
+                TRY_LOCK(::cs_main, lockMain);
+                if (lockMain) {
```

Comment:
Does it change something if  this check is moved above at the `m_context->connman` level ? You may save one more control flow.

Reply:
> Does it change something if this check is moved above at the `m_context->connman` level ? You may save one more control flow.

Why increase code block with locked `::cs_main`?

Reply:
> Does it change something if this check is moved above at the m_context->connman level ? You may save one more control flow.

Yes, it's a try lock. It's possible that we're not able to take the lock, and therefore that we're not able to get the node state stats, but we'd still get the node stats. If we move the try above the m_context->connman level, then if we couldn't get the cs_main lock, we wouldn't be able to get any stats.

Eventually, none of the statistics should require cs_main, so the try lock can be removed entirely.